### PR TITLE
Allow copyable GenericClientResponse structure and enhanced testing for inline/multi slices

### PR DESCRIPTION
**Pull request by hkadayam**

### CODE REVIEW

File: include/sisl/grpc/rpc_client.hpp [hunk 1]
```diff
@@ -166,21 +166,20 @@ class ClientRpcDataFuture : public ClientRpcDataInternal< ReqT, RespT > {
 class GenericClientResponse {
 public:
     GenericClientResponse() = default;
-    GenericClientResponse(grpc::ByteBuffer const& buf);
+    GenericClientResponse(grpc::ByteBuffer const& buf) : m_response_buf{buf} {}
```
Comment (szmyd) [inline include/sisl/grpc/rpc_client.hpp:169]: I'm surprised this isn't marked `explicit`; is that intentional?

**Review by szmyd — APPROVED** [review]
I'm not that familiar with `grpc::Slice` so will need to trust what's being done is correct. Otherwise :+1:

File: include/sisl/grpc/rpc_client.hpp (hunk 1 above)
Comment (raakella1) [reply include/sisl/grpc/rpc_client.hpp:169]: no, just an oversight. We can add the explicit specifier.

**Review by raakella1 — APPROVED** [review]
LG!

**Pull request merged**